virttest/libvirt_xml/vm_xml.py

```diff
@@ -2029,6 +2253,63 @@ def __init__(self, virsh_instance=base.virsh):
         super(VMCPUTuneXML, self).__init__(virsh_instance=virsh_instance)
         self.xml = '<cputune/>'
 
+    def update(self, attr_dict):
+        for attr, value in list(attr_dict.items()):
+            setattr(self, attr, value)
+
+    def get_memorytune(self):
+        xmltreefile = self.__dict_get__('xml')
+        try:
+            memorytune_root = xmltreefile.reroot('/memorytune')
+        except KeyError as detail:
+            raise xcepts.LibvirtXMLError(detail)
+        memorytunexml = MemoryTuneXML(virsh_instance=self.__dict_get__('virsh'))
+        memorytunexml.xmltreefile = memorytune_root
+        return memorytunexml
+
+    def set_memorytune(self, value):
+        if not issubclass(type(value), MemoryTuneXML):
+            raise xcepts.LibvirtXMLError("value must be a MemTuneXML \
+                                         or subclass")
+        xmltreefile = self.__dict_get__('xml')
+        # MemoryTuneXML root element is whole MemoryTune element tree
+        root = xmltreefile.getroot()
+        root.append(value.xmltreefile.getroot())
+        xmltreefile.write()
+
+    def del_memorytune(self):
+        xmltreefile = self.__dict_get__('xml')
+        element = xmltreefile.find('/memorytune')
+        if element is not None:
+            xmltreefile.remove(element)
+            xmltreefile.write()
+
+    def get_cachetune(self):
```

Comment:
cachetune can have more than one item ,how you handle this ?

https://libvirt.org/formatdomain.html

Reply:
It can handle more than one item. Memorytune also need to handle more than one item. They are the same.

Reply:
You can see the cfg file in https://github.com/autotest/tp-libvirt/pull/2108.